lib/usd/pxrUsdPreviewSurface/usdPreviewSurfacePlugin.h

```diff
@@ -18,35 +18,32 @@
 
 #include "api.h"
 
-#include <maya/MApiNamespace.h>
-
 #include <pxr/pxr.h>
+#include <pxr/base/tf/token.h>
+
+#include <maya/MApiNamespace.h>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
 /// \class PxrMayaUsdPreviewSurfacePlugin
 /// \brief Encapsulates plugin registration and deregistration of preview surface classes.
 ///
 /// Preview surface support requires plugin registration of node classes, node
-/// data, and draw support.  This class provides this service, including if
-/// multiple plugins that use preveiw surface are loaded: using reference
-/// counting, only the first registration and the last deregistration will
-/// be performed.  Note that because of Maya architecture requirements,
-/// deregistration will only be done if the deregistering plugin is the same as
-/// the registering plugin.  Otherwise, a warning is shown.
-
-class PxrMayaUsdPreviewSurfacePlugin
-{
-    public:
-        PXRUSDPREVIEWSURFACE_API
-        static MStatus initialize(MFnPlugin&);
+/// data, and draw support.  This class provides this service. Each client is expected
+/// to provide a separate typeName and typeId to ensure proper plugin registration.
+class PxrMayaUsdPreviewSurfacePlugin {
+public:
+    PXRUSDPREVIEWSURFACE_API
+    static MStatus initialize(MFnPlugin&, const MString&, MTypeId, const MString&);
 
-        PXRUSDPREVIEWSURFACE_API
-        static MStatus finalize(MFnPlugin&);
+    PXRUSDPREVIEWSURFACE_API
+    static MStatus finalize(MFnPlugin&, const MString&, MTypeId, const MString&);
```

Comment:
Would be good to include the parameter names here so it's clear which parameters are which.